config/routes.rb

```diff
@@ -2,7 +2,12 @@
 
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  # resources :users
+  root 'users#index'
+  get '/users', to: 'users#index'
+  get "/users/:id", to: "users#show"
+  get "/users/:id/posts", to: "posts#index"
+  get "/users/:id/posts/:id", to: "posts#show"
```

Comment:
- [ ] Your route definition here can prove problematic in the future. For example, the `id` parameter in this route `get "/users/:id/posts/:id", to: "posts#show"` does not accurately define which of the resources to which the id is referring.

  I think we can define these routes more accurately using the helper method provided by rails:
  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```
  This would more accurately define `id`, `post_id`, or `user_id` where necessary.